### PR TITLE
fix: [PIE-8316]: variables default issue refactor

### PR DESCRIPTION
**Pull request by vikrant-harness**
#### Screenshots

https://user-images.githubusercontent.com/106532291/221527665-71f5904c-7bf1-4194-9db2-41c0d040e813.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by vikrant-harness** [comment]
retrigger titlecheck

File: src/modules/10-common/components/ConfigureOptions/ConfigureOptionsUtils.ts [hunk 1]
```diff
@@ -181,7 +188,15 @@ export function parseInput(input: string): ParsedInput | null {
       const jexlMatch = fnArgs.match(JEXL_REGEXP)
 
       parsedInput[InputSetFunction.ALLOWED_VALUES] = {
-        values: jexlMatch ? null : fnArgs.split(','),
+        values: jexlMatch
+          ? null
+          : fnArgs.split(',').map(fnArg => {
+              const value = yamlParse(fnArg)
```
Comment (AshwinBhatkal-Harness) [inline src/modules/10-common/components/ConfigureOptions/ConfigureOptionsUtils.ts:194]: nit: can directly return this

File: src/modules/10-common/components/ConfigureOptions/ConfigureOptionsDialog.tsx [hunk 2]
```diff
@@ -84,7 +84,7 @@ export default function ConfigureOptionsDialog(props: ConfigureOptionsDialogProp
   const { showError } = useToaster()
   const { getString } = useStrings()
   const { NG_EXECUTION_INPUT } = useFeatureFlags()
-  const parsedValues = parseInput(input)
+  const parsedValues = parseInput(input, { variableType: type as any })
```
Comment (AshwinBhatkal-Harness) [inline src/modules/10-common/components/ConfigureOptions/ConfigureOptionsDialog.tsx:87]: add es lint comment to ignore

File: src/modules/10-common/components/ConfigureOptions/ConfigureOptionsUtils.ts [hunk 3]
```diff
@@ -181,7 +188,15 @@ export function parseInput(input: string): ParsedInput | null {
       const jexlMatch = fnArgs.match(JEXL_REGEXP)
 
       parsedInput[InputSetFunction.ALLOWED_VALUES] = {
-        values: jexlMatch ? null : fnArgs.split(','),
+        values: jexlMatch
```
Comment (AshwinBhatkal-Harness) [inline src/modules/10-common/components/ConfigureOptions/ConfigureOptionsUtils.ts:191]: add comments explaining this. Will help in the future

**Review by vkbansal-harness — DISMISSED** [review]

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

File: src/modules/10-common/components/ConfigureOptions/ConfigureOptionsUtils.ts [hunk 4]
```diff
@@ -181,7 +188,16 @@ export function parseInput(input: string): ParsedInput | null {
...
+          ? null
+          : // do not parse numbers if present under a string variables to support existing pipelines
+            fnArgs.split(',').map(fnArg => {
+              const value = yamlParse(fnArg)
+              if (variableType && variableType !== 'String') {
+                return value as unknown as string
```
Comment (rameshx) [inline src/modules/10-common/components/ConfigureOptions/ConfigureOptionsUtils.ts:197]: If `fnArg` should be parsed only when `variableType !== 'String'` then `const value = yamlParse(fnArg)` can be nested inside the `if` statement.

Comment (vikrant-harness) [reply src/modules/10-common/components/ConfigureOptions/ConfigureOptionsUtils.ts]: Done

**Review by rameshx — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**